Library/Paginate.swift

```diff
@@ -56,25 +56,25 @@ public func paginate <Cursor, Value: Equatable, Envelope, ErrorEnvelope, Request
     let isLoading = MutableProperty<Bool>(false)
 
     // Emits the last cursor when nextPage emits
-    let cursorOnNextPage = cursor.producer.ignoreNil().sampleOn(requestNextPage)
+    let cursorOnNextPage = cursor.producer.skipNil().sample(on: requestNextPage)
 
-    let paginatedValues = requestFirstPage
+    let paginatedValues: Signal<[Value], NoError> = requestFirstPage
       .switchMap { requestParams in
 
         cursorOnNextPage.map(Either.right)
-          .prefix(value: .left(requestParams))
+          .prefix(value: Either.left(requestParams))
           .switchMap { paramsOrCursor in
 
             paramsOrCursor.ifLeft(requestFromParams, ifRight: requestFromCursor)
-              .delay(AppEnvironment.current.apiDelayInterval, onScheduler: AppEnvironment.current.scheduler)
+              .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
               .on(
-                started: { [weak isLoading] _ in
+                starting: { [weak isLoading] _ in
```

Comment:
so ReactiveSwift introduced a new producer lifecycle state, [`starting`](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Sources/SignalProducer.swift#L1360), which is called before anything happens with the producer, where as `started` is called after the producer has started doing things.

this difference caused some tests to fail for our time-based tests that use 0 seconds for testing. the fix is easy tho!